src/react-hook/react-hook.tsx

```diff
@@ -19,11 +19,11 @@ export function useConfig() {
       "ModuleNameContext has not been provided. This should come from openmrs-react-root-decorator"
     );
   }
-  if (!config) {
+  if (typeof config === "undefined") {
     // React will prevent the client component from rendering until the promise resolves
     throw Config.getConfig(moduleName).then(res => {
       config = res;
-      //      setConfig(res);
+      return config;
```

Comment:
I don't think that returning the config here does anything. This is a React suspense promise, and the return value is not used for anything afaik.